pyproject.toml

```diff
@@ -32,6 +32,7 @@ geopy = "^2.1.0"
 timezonefinder = "^6.0.1"
 backoff = "^1.11.1"
 chardet = "^5"
+openai = "^0.28.1"
```

Comment:
How heavyweight is this dependency? Would it be preferable to make it an optional extra instead of a hard requirement?

Reply:
It doesn't look crazy: https://github.com/openai/openai-python/blob/main/pyproject.toml#L10
However, I think that making it optional is an extra safeguard to make people aware of using it
And, I also noticed that there is a stable version!